lib/msal-core/src/messaging/MessageHelper.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { WindowUtils } from "./../utils/WindowUtils";
+import { MessageCache } from "./MessageCache";
+import { Logger } from "./../Logger";
+import { MessageDispatcher } from "./MessageDispatcher";
+
+export enum MessageType {
+    REDIRECT_REQUEST = "REDIRECT_REQUEST",
+    HASH = "HASH",
+    URL_TOP_FRAME = "URL_TOP_FRAME",
+    URL_NAVIGATE = "URL_NAVIGATE"
+};
+
+export enum WindowType {
+    TOP_FRAME,
+    IFRAME
+};
+
+export type PAYLOAD = {
+    type: MessageType,
+    data?: string
+};
+
+export class MessageHelper {
+
+    /**
+     * returns the current window type: Top Frame app or Iframed app
+     */
+    static currentWindow(): WindowType {
+        if(WindowUtils.isWindowOnTop()) {
+            return WindowType.TOP_FRAME;
+        }
+        else if(WindowUtils.isInIframe()) {
```

Comment:
Don't need `else` since previous blocks are returning.

Reply:
Of course. Done.